--- libmpdclient-2.25/src/song.c.orig	2026-06-17 11:58:42.000000000 +0200
+++ libmpdclient-2.25/src/song.c	2026-06-19 19:50:40.377520115 +0200
@@ -219,7 +219,8 @@ mpd_song_dup(const struct mpd_song *song
 		} while (src_tag != NULL);
 	}
 
-	ret->real_uri = strdup(song->real_uri);
+	if (song->real_uri != NULL)
+		ret->real_uri = strdup(song->real_uri);
 	ret->duration = song->duration;
 	ret->duration_ms = song->duration_ms;
 	ret->start = song->start;
